receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,306 @@
+		return filterClusters(clusters, include, true), nil
+	// exclude is initialized
+	case len(exclude) > 0 && len(include) == 0:
+		return filterClusters(clusters, exclude, false), nil
+	// both are initialized
+	default:
+		return nil, errors.New("both Include and Exclude clusters configured")
+	}
+}
+
+func (s *logsReceiver) collectClusterLogs(clusters []mongodbatlas.Cluster, project ProjectConfig, pc ProjectContext) {
+	for _, cluster := range clusters {
+		hostnames := parseHostNames(cluster.ConnectionStrings.Standard, s.log)
+		for _, hostname := range hostnames {
+			s.collectLogs(pc, "mongodb.gz", hostname, cluster.Name)
+			s.collectLogs(pc, "mongos.gz", hostname, cluster.Name)
+
+			if project.EnableAuditLogs {
+				s.collectAuditLogs(pc, "mongodb-audit-log.gz", hostname, cluster.Name)
+				s.collectAuditLogs(pc, "mongos-audit-log.gz", hostname, cluster.Name)
+			}
+		}
+	}
+}
+
+func filterClusters(clusters []mongodbatlas.Cluster, clusterNames []string, include bool) []mongodbatlas.Cluster {
+	// Arrange cluster names for quick reference
+	clusterNameSet := map[string]struct{}{}
+	for _, clusterName := range clusterNames {
+		clusterNameSet[clusterName] = struct{}{}
+	}
+
+	var filtered []mongodbatlas.Cluster
+	for _, cluster := range clusters {
+		if _, ok := clusterNameSet[cluster.Name]; (!ok && !include) || (ok && include) {
+			filtered = append(filtered, cluster)
+		}
+	}
+	return filtered
+}
+
+func (s *logsReceiver) getHostLogs(groupID, hostname, logName string, start, end time.Time) ([]model.LogEntry, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName, start, end)
+	if err != nil {
+		return nil, err
+	}
+	// Pass this into a gzip reader for decoding
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Logs are in JSON format so create a JSON decoder to process them
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.LogEntry, 0)
+	for {
+		var entry model.LogEntry
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *logsReceiver) getHostAuditLogs(groupID, hostname, logName string, start, end time.Time) ([]model.AuditLog, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName, start, end)
+	if err != nil {
+		return nil, err
+	}
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.AuditLog, 0)
+	for {
+		var entry model.AuditLog
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *logsReceiver) collectLogs(pc ProjectContext, logName, hostname, clusterName string) {
```

Comment:
nit: since we're passing hostname and logName through several functions, let's keep the order consistent. The client use hostname before logName, so let's do that here and on `collectAuditLogs` too.